assets/src/scripts/components/water-level-table/index.js

```diff
@@ -0,0 +1,84 @@
+/* global List */
+
+import { select } from 'd3-selection';
+import { createStructuredSelector } from 'reselect';
+
+import { link } from 'ngwmn/lib/d3-redux';
+import { getSiteWaterLevels} from 'ngwmn/services/state/index';
+
+import { isTableRendered, renderTable } from './state';
+
+const COLUMN_HEADINGS = [
+    'Sample Time',
+    'Original Value',
+    'Original Unit',
+    'Accuracy Value',
+    'Accuracy Unit',
+    'Data Provided by',
+    'Depth to water, feet below land surface',
+    'Water level in feet relative to NAVD88'
+];
+
+/*
+ * Draws the waterLevels data into table
+ * @param {D3 node} table
+ * @param {Array of Object} waterLevels
+ * @return {Object} - Return the tbody element for the table
+ */
+const drawTableBody = function(table, waterLevels, tbody) {
+    tbody = tbody || table
+        .append('tbody')
+        .classed('list', true);
+    const samples = waterLevels.samples.reverse() || [];
+    const valueNames = ['time', 'originalValue', 'unit', 'accuracyValue', 'accuracyUnit',
+            'sourceCode', 'fromLandsurfaceValue', 'fromDatumValue'];
+    const item = valueNames.reduce(function(total, name) {
+        return `${total}<td class="${name}"></td>`;
+    }, '');
+    const options = {
+        valueNames: valueNames,
+        item: `<tr>${item}</tr>`,
+        page: 30,
+        pagination: true
+    };
+    new List('water-levels-div', options, samples);
+
+    return tbody;
+};
+/*
+ * Renders the water level table
+ * @param  {Object} store   Redux store
+ * @param  {Object} node    DOM node to draw graph into
+ * @param  {Object} options {agencycode, siteid} of site to draw
+ */
+export default function(store, node, {agencycd, siteid}) {
+    const component = select(node);
+    component.select('button').on('click', () => {
+        store.dispatch(renderTable());
+    });
+    const table = component
+        .select('#water-levels-div')
+            .append('table')
+        .attr('id', 'water-levels-table')
+            .classed('usa-table', true);
+    component.select('#water-levels-div')
+        .append('ul')
+        .classed('pagination', true);
+
+    table.append('thead')
+        .append('tr')
+            .selectAll('th')
+            .data(COLUMN_HEADINGS).enter()
+            .append('th')
+                .text((col) => col);
+
+    table.call(link(store, (elem, {isRendered, waterLevels}) => {
+        // Add code to rendered
+        if (isRendered) {
+            drawTableBody(elem, waterLevels);
+        }
+    }, createStructuredSelector({
+        isRendered: isTableRendered,
```

Comment:
I feel like we don't need to track this in the Redux state - we could re-render the table whenever the waterLevels have changed, just replacing the body. If water levels are undefined, a loading indicator could be drawn.

Reply:
I did it this way so that the data wasn't rendered until the accordian was opened.

Reply:
Ah right, sorry makes sense now!